runtime/src/software_interrupt.c

```diff
@@ -78,16 +37,22 @@ extern pthread_t *runtime_worker_threads;
  * Private Static Inlines *
  *************************/
 
+static inline void
+defer_sigalrm()
+{
+	atomic_fetch_add(&deferred_sigalrm, 1);
+}
+
 /**
  * A POSIX signal is delivered to only one thread.
  * This function broadcasts the sigalarm signal to all other worker threads
  */
 static inline void
-sigalrm_propagate_workers(siginfo_t *signal_info)
+propagate_sigalrm(siginfo_t *signal_info)
 {
 	/* Signal was sent directly by the kernel, so forward to other threads */
 	if (signal_info->si_code == SI_KERNEL) {
-		atomic_fetch_add(&software_interrupt_SIGALRM_kernel_count, 1);
+		software_interrupt_counts_sigalrm_kernel_increment();
```

Comment:
Can take this out of the conditionals as it is the first op in the `if` and the `else`.

Reply:
They're actually different functions. One is `*_sigalrm_kernel_increment` and the other is `*_sigalrm_thread_increment`. It help track the amount of times a worker receives the original SIGALRM from the kernel versus a propagated SIGALRM from another thread. The kernel count generally shows that the lowest numbered worker generally gets most of the SIGALRMs form the kernel. The user count shows how efficient the "smart triage" logic is. It seems to reduce interrupts by ~95% for all other core.